Add Tutorial(1. index postgres; 2. setup superset preview client) for the doc (#413)
lyft/amundsen: docs/tutorials/data-preview-with-superset.md

# How to setup a preview client with Apache Superset

> In the previous [tutorial](docs/tutorials/index-postgres.md), we talks about how to index the table metadata

Reviewer: ```suggestion
In the previous [tutorial](docs/tutorials/index-postgres.md), we talked about how to index the table metadata
```

> for a postgres database. In this tutorial, we will walk through how to config data preview for this `films` table

Reviewer: ```suggestion
for a postgres database. In this tutorial, we will walk through how to configure data preview for this `films` table
```

    
    # Initialize the database
    superset db upgrade
    
    # Create an admin user (you will be prompted to set a username, first and last name before setting a password)
    $ export FLASK_APP=superset
    superset fab create-admin
    
    # Load some data to play with
    superset load_examples
    
    # Create default roles and permissions
    superset init
    
    # To start a development web server on port 8088, use -p to bind to another port
    superset run -p 8088 --with-threads --reload --debugger
   ```
   
   Once setup properly, you could view the superset UI as following:
   ![](../img/tutorials/superset-welcome.png)
   
2. We need to add the postgres database to superset as the following:
![](../img/tutorials/superset-add-db.png)

3. We could verify the content of the `films` table using superset's sqlab feature:
![](../img/tutorials/superset-sqllab-verify.png)

4. Next, We need to build a preview client following this [guide](../frontend/docs/examples/superset_preview_client.md) 
and the [example client code](https://github.com/lyft/amundsenfrontendlibrary/blob/master/amundsen_application/base/examples/example_superset_preview_client.py).

> There are couples of things keep in mind:

Reviewer: ```suggestion
There are a couple of things keep in mind:
```

Reviewer 2: Even: things **to** keep in mind

    - We could start with an unauthenticated Superset([example superset config](https://gist.github.com/feng-tao/b89e6faf7236372cef70a44f13615c39)), 

>     but in production, we will need to send the impersonate info to Superset

Reviewer: So the impersonation support is in open source Superset as well?

Author: i check which Lyft superset override that endpoint to do personal authorization with TOM, while OSS superset does it a similar but rely on FAB to do impersonation API security check.

    to properly verify whether the given user could view the data.

>     - When we build the client, we could need configure the database id instead of database name when send the request to superset.

Reviewer: A little confused by this statement -- in the open source version does it work if you send the database name in the request?

Author: still requires db id, let me clarify a bit.

> 5. Once we configure the preview client, put it in the frontend service entry point and restart the frontend.

Reviewer: Can link here for examples of using the entry point. https://github.com/lyft/amundsenfrontendlibrary/blob/master/docs/configuration.md#python-entry-points
